libs/ledger/src/lib/ledger-export-form.spec.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  act,
+  render,
+  screen,
+  waitFor,
+  fireEvent,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { MockedProvider } from '@apollo/client/testing';
+import { LedgerExportForm } from './ledger-export-form';
+import { AssetsDocument } from '@vegaprotocol/assets';
+import { formatForInput, toNanoSeconds } from '@vegaprotocol/utils';
+
+const VEGA_URL = 'https://vega-url.co.uk/querystuff';
+const mockEnvironment = jest.fn(() => VEGA_URL);
+jest.mock('@vegaprotocol/environment', () => ({
+  useEnvironment: jest.fn(() => mockEnvironment()),
+}));
+
+const mockResponse = {
+  headers: { get: jest.fn() },
+  blob: () => '',
+};
+global.fetch = jest.fn().mockResolvedValue(mockResponse);
+
+const assetMock = {
+  request: {
+    query: AssetsDocument,
+  },
+  result: {
+    data: {
+      assetsConnection: {
+        edges: [
+          {
+            node: {
+              id: 'asset-id',
+              symbol: 'symbol asset-id',
+            },
+          },
+          {
+            node: {
+              id: 'asset-id-2',
+              symbol: 'symbol asset-id-2',
+            },
+          },
+        ],
+      },
+    },
+  },
+};
+
+describe('LedgerExportLink', () => {
+  const partyId = 'partyId';
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+  beforeAll(() => {
+    jest.useFakeTimers().setSystemTime(new Date('2023-08-10T10:10:10.000Z'));
+  });
+  afterAll(() => {
+    jest.useRealTimers();
+  });
+
+  it('should be properly rendered', async () => {
+    render(
+      <MockedProvider mocks={[assetMock]}>
+        <LedgerExportForm partyId={partyId} />
+      </MockedProvider>
+    );
+    await waitFor(() => {
+      expect(
```

Comment:
chore (blocking): This can be more succinct with expect(await screen.findByRole...

Reply:
done